Deployments with deprecated runtimes (#3553)
astronomer/docs: astro/upgrade-runtime.md

To stay up to date on the latest versions of Astro Runtime, see [Astro Runtime release notes](runtime-release-notes.md). For more information on Astro Runtime versioning and support, see [Astro Runtime versioning and lifecycle policy](runtime-version-lifecycle-policy.md). For a full collection of Astro Runtime Docker images, go to the [Astro Runtime repository on Quay.io](https://quay.io/repository/astronomer/astro-runtime?tab=tags).

### Run a deprecated Astro Runtime

<HostedBadge/>

> If you're migrating to Astro from OSS Airflow, Software, or Astro Hybrid, where your Deployments run using an older version of Airflow or a deprecated version of the Astro Runtime, you can now use stepwise migration to complete your migration in stages. First, you can create Deployments with a deprecated version of the Astro Runtime using the Astro API to move to an Astro Hosted environment. Then, you can upgrade your code to the most up-to-date version of the Astro Runtime.

Author: ```suggestion
If you're migrating to Astro from OSS Airflow or Astro Hybrid, where your Deployments run using an older version of Airflow or a deprecated version of the Astro Runtime, you can now use stepwise migration to complete your migration in stages. First, you can create Deployments with a deprecated version of the Astro Runtime using the Astro API to move to an Astro Hosted environment. Then, you can upgrade your code to the most up-to-date version of the Astro Runtime.
```

When you create a Deployment with a deprecated Astro Runtime version, Astronomer Support still follows the [Astro Runtime maintenance policy](runtime-version-lifecycle-policy.md#astro-runtime-maintenance-policy), and might advise you to upgrade your runtime to the latest version of Astro to resolve performance issues or bugs.

> :::note

Reviewer: ```suggestion
:::info
```

> There are some versions of deprecated Astro Runtime that included undesirable behavior or bugs in the version of either Airflow or Astro Runtime. If you try to use one of these versions when creating a Deployment with a deprecated Runtime, the Astro API returns a failure. See the [Version upgrade considerations](#version-upgrade-considerations) for additional information when choosing a Runtime version to use.

Reviewer: ```suggestion
When you choose a deprecated version of Astro Runtime, Astronomer recommends always choosing the latest patch version that includes your desired Airflow version. Some earlier patch versions of Astro Runtime include unexpected behavior and can't be used even as a deprecated version. For more information about which versions are affected, see the [Version upgrade considerations](#version-upgrade-considerations).
```

:::

> 1. Contact [Astronomer Suppor](astro-support.md) to allow your Deployments to run deprecated versions of the Astro Runtime.

Reviewer: ```suggestion
1. Contact [Astronomer support](astro-support.md) and request the ability to run deprecated versions of Astro Runtime.
```

> 2. Create either a Dedicated or Standard Deployment using the [Astro API](https://docs.astronomer.io/api/platform-api-reference/deployment/create-deployment), and define the `astroRuntimeVersions` using the deprecated Runtime version you want to use.

Reviewer: ```suggestion
2. Create a Deployment using the [Astro API](https://docs.astronomer.io/api/platform-api-reference/deployment/create-deployment). In your request, specify your deprecated Astro Runtime version using the `astroRuntimeVersions` property.
```

> 3. (Optional) After you create the Deployment, you can manage settings and other features using the [Astro UI](deployment-settings.md).

Reviewer: ```suggestion

After you create the Deployment, you can manage the Deployment using any interface, including the Astro CLI and [Astro UI](deployment-settings.md).
```
